### PR TITLE
Add errormessage to  coldatomjob

### PR DESCRIPTION
**Pull request by LaurinFischer**
### Summary
This PR implements an `error_message` method to the `ColdAtomJob` class which enables users to read off the error message sent by a remote backend as `job.error_message()`.

### Details and comments
Not sure how to include this in tests, as this is only relevant for interactions with a remote backend. 
As of now, we have omitted tests for the requests that interact with remote backends.

### CODE REVIEW

File: qiskit_cold_atom/providers/cold_atom_job.py [hunk 1]
```diff
@@ -135,6 +135,28 @@ def status(self):
 
         return status
 
+    def error_message(self):
```
Comment (eggerdj) [inline qiskit_cold_atom/providers/cold_atom_job.py:138]:
```suggestion
    def error_message(self) -> Optional[str]:
```

File: qiskit_cold_atom/providers/cold_atom_job.py [hunk 2]
```diff
@@ -135,6 +135,28 @@ def status(self):
...
+            params={
+                "json": json.dumps(status_payload),
+                "username": self.user,
+                "password": self.token,
+            },
+        )
+
+        error_message = r.json()["error_message"]
+
+        return error_message
```
Comment (eggerdj) [inline qiskit_cold_atom/providers/cold_atom_job.py:158]:
```suggestion
        return r.json().get("error_message", None)
```
This should be more robust: covers the case where there was no error or if the backend did not define one.

**Review by eggerdj — CHANGES_REQUESTED** [review]

File: qiskit_cold_atom/providers/cold_atom_job.py (hunk 2 above)
Comment (LaurinFischer) [reply qiskit_cold_atom/providers/cold_atom_job.py:158]: I was wondering how best to do this, thanks.

**Review by eggerdj — APPROVED** [review]
LGTM.

**Pull request merged**